src/gazebo_mavlink_interface.cpp

```diff
@@ -554,18 +554,16 @@ void GazeboMavlinkInterface::Load(physics::ModelPtr _model, sdf::ElementPtr _sdf
 // This gets called by the world update start event.
 void GazeboMavlinkInterface::OnUpdate(const common::UpdateInfo&  /*_info*/) {
 
-  std::unique_lock<std::mutex> lock(last_imu_message_mutex_);
+  std::unique_lock<std::mutex> lock(imu_received_mutex_);
 
-  if (previous_imu_seq_ > 0) {
-    while (previous_imu_seq_ == last_imu_message_.seq() && IsRunning()) {
-      last_imu_message_cond_.wait_for(lock, std::chrono::milliseconds(10));
+  if (imu_received_once_) {
+    while (!imu_received_ && IsRunning()) {
+      imu_received_cond_.wait_for(lock, std::chrono::milliseconds(10));
     }
   }
-
-  previous_imu_seq_ = last_imu_message_.seq();
-
+  imu_received_ = false;
```

Comment:
If we are considering the usecase of a multi imu scenario that the lockstep is based on a certain imu, the `imu_received_` should only increment when the primary imu has been updated, not all of them

Reply:
Yes i will make this on the multi IMU PR